app/lib/actions/middleware/propagate_output.rb

```diff
@@ -0,0 +1,14 @@
+module Actions
+  module Middleware
+    class PropagateOutput < Dynflow::Middleware
+      def run(*args)
+        pass(*args)
+        if action.input.keys.include?('subaction_output') && action.input['subaction_output']
+          action.input['subaction_output'].each do |key, value|
+            self.action.output[key] = value
+          end
+        end
```

Comment:
Can you delete the 'subaction_output' on the input after this runs?    The tasks show up twice (on the input and output) without us clearing it.